app/src/main/java/com/example/dispatcher/presentation/search/view/SearchFragment.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.dispatcher.presentation.search.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.dispatcher.R
+import com.example.dispatcher.databinding.FragmentSearchBinding
+import com.example.dispatcher.presentation.homepage.view.adapter.ArticleAdapter
+import com.example.dispatcher.presentation.homepage.view.adapter.TopSpacingItemDecoration
+import com.example.dispatcher.presentation.homepage.viewModel.ArticlesViewModel
+import com.example.dispatcher.presentation.main.MainViewModel
+import com.example.dispatcher.presentation.search.view.adapter.SearchHistoryAdapter
+import com.example.dispatcher.presentation.search.viewModel.SearchViewModel
+
+class SearchFragment : Fragment(R.layout.fragment_search) {
+
+    private var _binding: FragmentSearchBinding? = null
+    private val binding get() = _binding!!
+
+    private val mainViewModel: MainViewModel by activityViewModels()
+    private val articlesViewModel: ArticlesViewModel by viewModels()
+    private val searchViewModel: SearchViewModel by activityViewModels()
+
+    private lateinit var articleAdapter: ArticleAdapter
+    private lateinit var searchHistoryAdapter: SearchHistoryAdapter
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentSearchBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.searchView.setViewModel(mainViewModel)
+        binding.searchView.setArticlesViewModel(articlesViewModel)
+        binding.searchView.setSearchViewModel(searchViewModel)
+        binding.filter.setViewModel(mainViewModel)
+
+        binding.resultsLayout.visibility = View.GONE
+        binding.notFoundLayout.visibility = View.GONE
+        observeToSearchArticles()
+        addAdapter()
+        addSearchHistoryAdapter()
+        observeSearchHistory()
+
+        setListeners()
+    }
+
+    private fun observeToSearchArticles() {
+        articlesViewModel.searchArticlesLiveData.observe(viewLifecycleOwner) { articles ->
+
+            if (articles != null) {
+                if (articles.size != 0) {
+                    articleAdapter.submitList(articles)
```

Comment:
please notice that you made the adapter a `lateinit` and you call it in here, but the init is actually after this function - this cause a problem